File: lab4/s2/server/src/main/java/ru/uvuv643/dto/human/PairTeamHumanDto.java
package ru.uvuv643.dto.human;

import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;

@JacksonXmlRootElement(localName = "pairTeamHumanDto") // Root XML tag for individual pair
public class PairTeamHumanDto {

        @JacksonXmlProperty(localName = "humanBeingId") // Tag for humanBeingId field
        private int humanBeingId;

        @JacksonXmlProperty(localName = "teamId") // Tag for teamId field
        private int teamId;

        // Default Constructor
        public PairTeamHumanDto() {
        }

        // Constructor
        public PairTeamHumanDto(int humanBeingId, int teamId) {
                this.humanBeingId = humanBeingId;
                this.teamId = teamId;
        }

        // Getters and Setters
        public int getHumanBeingId() {
                return humanBeingId;
        }

        public void setHumanBeingId(int humanBeingId) {
                this.humanBeingId = humanBeingId;
        }

        public int getTeamId() {
                return teamId;
        }

        public void setTeamId(int teamId) {
                this.teamId = teamId;
        }
}

File: lab1/s2/src/main/kotlin/ru/uvuv643/soa/api/v1/dto/human/response/PairTeamHumanDto.kt
package ru.uvuv643.soa.api.v1.dto.human

import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElement
import jakarta.xml.bind.annotation.XmlRootElement

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
data class PairTeamHumanDto (

        @XmlElement(name = "humanBeingId")
        val humanBeingId : Int,

        @XmlElement(name = "teamId")
        val teamId : Int,

) {
        constructor() : this(0, 0)
}